### ISSUE TITLE
Join table with other columns

### ISSUE THREAD

**Issue by deafwolf** [vattle/sqlboiler#81]
In function setIsJoinTable

```
	if t.PKey == nil || len(t.PKey.Columns) != 2 || len(t.FKeys) < 2 || len(t.Columns) > 2 {
		return
	}
```

Is the `len(t.Columns) > 2` nessary? I think maybe the join table need created_at/updated_at

**Comment by nullbio** [comment]
@aarondl Any idea? I'm wondering if `len(t.PKey.Columns) != 2` should actually be `len(t.PKey.Columns) < 2` in case someone has a composite key with more than 2 columns.

**Comment by deafwolf** [comment]
I think removing the `len(t.Columns) > 2` is better, so that we can add created_at/updated_at.
We only want to add these two columns, but I'm afraid this change will break the join table's behavior(more non automatical columns need filling value).

**Comment by aarondl** [comment]
Hey @deafwolf. Unfortunately we can't simply remove that condition. When a join table is detected by that function, we don't even generate a model for it or anything to keep it completely transparent (convenient) for users. As soon as the join table has another column, we can't completely remove the model generation and helper generation for it because the system doesn't know how to remove a table with more columns than that.

What I would suggest is a hybrid approach where we can have "pure" and "impure" join tables. A pure join table will have it's model completely removed which is the existing functionality. An impure join table which is defined by >=3 columns + primary key composed of two foreign keys will still have a model generated, but in addition to that have relationship helpers generated as if it were a join table so that querying through it still is simple.

I think this approach is correct but unfortunately it's a lot of work and since there are plenty of workarounds (use raw queries or manually manage the relationships of the join table) we probably won't be jumping on this immediately. We will keep it in the backlog though, and thank you for the suggestion.

**Comment by deafwolf** [comment]
@aarondl Yes, you are right.

What I need is when joining pilots and languages with impure join table, I can call pilot.Languages() directly. I have tried use AUTO_INCREMENT as primary key of pilot_language, sqlboiler gen a PilotLanguage and I can only call pilot.PilotLanguage().

I think maybe this feature isn't very important. Please think more if sqlboiler really need to support it.

Thanks.

**Comment by aarondl** [comment]
Yeah, your description is consistent with what I mentioned above. Well keep it in the backlog for sure because I do think there's some value in it. Thanks again :)

**Comment by deafwolf** [comment]
👍
